analysis/src/Hint.ml

```diff
@@ -156,30 +141,34 @@ let codeLens ~path ~debug =
    in
    let {Res_driver.parsetree = structure} = parser ~filename:path in
    iterator.structure iterator structure |> ignore);
-  !lenses
-  |> List.filter_map (fun (range : Protocol.range) ->
-         match Cmt.fullFromPath ~path with
-         | None -> None
-         | Some full -> (
-           match
-             References.getLocItem ~full
-               ~pos:(range.start.line, range.start.character + 1)
-               ~debug
-           with
-           | Some {locType = Typed (_, typeExpr, _)} ->
-             Some
-               (Protocol.stringifyCodeLens
-                  {
-                    range;
-                    command =
-                      Some
-                        {
-                          (* Code lenses can run commands. An empty command string means we just want the editor
-                             to print the text, not link to running a command. *)
-                          command = "";
-                          (* Print the type with a huge line width, because the code lens always prints on a
-                             single line in the editor. *)
-                          title = typeExpr |> Shared.typeToString ~lineWidth:400;
-                        };
-                  })
-           | _ -> None))
+  match Cmt.fullFromPath ~path with
```

Comment:
I think this is great. Question for the future: does it make sense to have some sort of cache inside of `Cmt.fullFromPath` directly?

Reply:
The change looks good. Perhaps it's possible to rename `fullFromPath` to something that makes it clear it's an expensive operation. E.g. `loadFullCmtFromPath`.

Reply:
Done in 0f260537faf103aca33519ce1da93fae0a1e2479